chia/wallet/wallet.py

```diff
@@ -265,27 +266,23 @@ async def select_coins(self, amount, exclude: List[Coin] = None) -> Set[Coin]:
         unspent: List[WalletCoinRecord] = list(
             await self.wallet_state_manager.get_spendable_coins_for_wallet(self.id())
         )
-        sum_value = 0
-        used_coins: Set = set()
-
-        # Use older coins first
-        unspent.sort(reverse=True, key=lambda r: r.coin.amount)
-
         # Try to use coins from the store, if there isn't enough of "unused"
         # coins use change coins that are not confirmed yet
         unconfirmed_removals: Dict[bytes32, Coin] = await self.wallet_state_manager.unconfirmed_removals_for_wallet(
             self.id()
         )
-        for coinrecord in unspent:
-            if sum_value >= amount and len(used_coins) > 0:
-                break
+
+        sum_value = 0
```

Comment:
Sum value is not used

Reply:
Changed it and it is now used to check if we have enough coins.